app/controllers/registrations_controller.rb

```diff
@@ -0,0 +1,57 @@
+class RegistrationsController < Devise::RegistrationsController
+  before_action :configure_account_update_params, only: [:update]
+  prepend_before_action :authenticate_scope!, only: %i[edit update destroy create new]
+
+  #GET /resource/edit
+  def edit
+    render :edit
+  end
+
+  # PUT /resource
+  def update
+    #self.resource = resource_class.to_adapter.get!(send(:"current_#{resource_name}").to_key)
+    #prev_unconfirmed_email = resource.unconfirmed_email if resource.respond_to?(:unconfirmed_email)
+    self.resource = send(:"authenticate_#{resource_name}!", force: true)
+    resource_updated = update_resource(resource, account_update_params)
+    yield resource if block_given?
+    if resource_updated
+      set_flash_message_for_update(resource, "prev_unconfirmed_email")
+      bypass_sign_in resource, scope: resource_name if sign_in_after_change_password?
+
+      respond_with resource, location: after_update_path_for(resource)
+    else
+      clean_up_passwords resource
+      set_minimum_password_length
+      respond_with resource
+    end
+  end
+
+  def update_resource(resource, params)
+    if params.has_key?(:session_only) && params[:session_only]
+      resource.update_without_password(params)
+    else
+      resource.update_with_password(params)
+    end
+  end
+
+  # protected
+
+  def authenticate_scope!
+    self.resource = send(:"authenticate_#{resource_name}!", force: true)
```

Comment:
I know Ruby is awesome and all, but not so keen on dynamically created method calls